### PR TITLE
Core: use ManifestFiles.open when possible

### PR DESCRIPTION
**Pull request by dramaticlly**
walking through my old discarded PR and found somewhere we can improve

If you want to take a look ? @szehon-ho @RussellSpitzer

### CODE REVIEW

**Review by hsiang-c — APPROVED** [review]
LGTM

File: core/src/main/java/org/apache/iceberg/BaseFilesTable.java [hunk 1]
```diff
@@ -166,7 +166,6 @@ public CloseableIterable<StructLike> rows() {
       if (readableMetricsField == null) {
         return CloseableIterable.transform(files(projection), file -> (StructLike) file);
       } else {
-
```
Comment (szehon-ho) [inline core/src/main/java/org/apache/iceberg/BaseFilesTable.java:169]: Nit: revert this change?

**Review by szehon-ho — APPROVED** [review]
great catch

File: core/src/main/java/org/apache/iceberg/BaseFilesTable.java (hunk 1 above)
Comment (szehon-ho) [reply core/src/main/java/org/apache/iceberg/BaseFilesTable.java:169]: oh yea it is definitely subjective, i remember vaguely Anton suggest some newlines after else for better readability at some point for my prs (not sure for this particular line or not).  Id say just remove it to minimize the change?

Comment (dramaticlly) [reply core/src/main/java/org/apache/iceberg/BaseFilesTable.java]: thank you for the context, reverted!

**Review by RussellSpitzer — APPROVED** [review]

**Comment by RussellSpitzer** [comment]
Thanks @dramaticlly for the fix - Thanks @szehon-ho, @singhpk234  and @hsiang-c for review
